src/test/sequence_test.cpp

```diff
@@ -0,0 +1,33 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <memory>
+#include <mutex>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "../sequence.hpp"
+
+TEST(Sequence, MovedMutexNullified) {
```

Comment:
I think adding ULT for  updateMemoryState is crucial here. Also we should check last modification time was updated.